compiler/src/iree/compiler/Codegen/Common/ConfigTrackingCanonicalizer.cpp

```diff
@@ -90,19 +90,18 @@ struct ConfigTrackingCanonicalizerPass final
   }
 
   void runOnOperation() override {
+    config.listener = &listener;
+
     // Canonicalization is best-effort. Non-convergence is not a pass failure.
-    ConfigTrackingListener listener;
-    {
-      config.listener = &listener;
-      LogicalResult didConverge =
-          applyPatternsAndFoldGreedily(getOperation(), *patterns, config);
-      if (this->testConvergence && failed(didConverge)) {
-        getOperation()->emitError("Canonicalizer failed to converge");
-        return signalPassFailure();
-      }
+    LogicalResult didConverge =
```

Comment:
Here, I think it should be enough to set `config.listener` to nullptr after we run `applyPatterns...`